src/tables/cpal.js

```diff
@@ -44,5 +50,263 @@ function makeCpalTable({ version = 0, numPaletteEntries = 0, colorRecords = [],
+
+    if (index > cpalTable.numPaletteEntries) {
+        throw new Error(`Color index out of range (numPaletteEntries: ${cpalTable.numPaletteEntries}, index: ${index})`);
+    }
+
+    const lookupIndex = cpalTable.colorRecordIndices[palette] + index;
+    if (lookupIndex > cpalTable.colorRecords) {
+        throw new Error(`Color index out of range (colorRecords.length: ${cpalTable.colorRecords.length}, lookupIndex: ${lookupIndex})`);
+    }
+
+    const color = parseCPALColor(cpalTable.colorRecords[lookupIndex]);
+    if(colorFormat === 'bgra') {
+        return color;
+    }
+    return formatColor(color, colorFormat);
+}
+
+function toHex(d) {
+    return ('0' + parseInt(d).toString(16)).slice(-2);
+}
+
+function rgbToHSL(bgra) {
+    const r = bgra.r/255;
+    const g = bgra.g/255;
+    const b = bgra.b/255;
+    const max = Math.max(r, g, b), min = Math.min(r, g, b);
+    let h, s, l = (max + min) / 2;
+
+    if (max === min) {
+        h = s = 0;
+    } else {
+        const d = max - min;
+        s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
+        switch (max) {
+            case r: h = (g - b) / d + (g < b ? 6 : 0); break;
+            case g: h = (b - r) / d + 2; break;
+            case b: h = (r - g) / d + 4; break;
+        }
+        h /= 6;
+    }
+
+    return {
+        h: h * 360,
+        s: s * 100,
+        l: l * 100
+    };
+}
+
+function hslToRGB(hsla) {
+    let { h, s, l, a } = hsla;
+    h = h % 360;
+    s /= 100;
+    l /= 100;
+
+    const c = (1 - Math.abs(2 * l - 1)) * s;
+    const x = c * (1 - Math.abs((h / 60) % 2 - 1));
+    const m = l - c / 2;
+
+    let r = 0, g = 0, b = 0;
+
+    if (0 <= h && h < 60) {
+        r = c; g = x; b = 0;
+    } else if (60 <= h && h < 120) {
+        r = x; g = c; b = 0;
+    } else if (120 <= h && h < 180) {
+        r = 0; g = c; b = x;
+    } else if (180 <= h && h < 240) {
+        r = 0; g = x; b = c;
+    } else if (240 <= h && h < 300) {
+        r = x; g = 0; b = c;
+    } else if (300 <= h && h <= 360) {
+        r = c; g = 0; b = x;
+    }
+  
+    return {
+        r: Math.round((r + m) * 255),
+        g: Math.round((g + m) * 255),
+        b: Math.round((b + m) * 255),
+        a
+    };
+}
+
+function bgraToRaw(color) {
+    return parseInt(`0x${toHex(color.b)}${toHex(color.g)}${toHex(color.r)}${toHex(color.a * 255)}`, 16);
+}
+
+function parseColor(color, targetFormat = 'hexa') {
+    const returnRaw = (targetFormat == 'raw' || targetFormat == 'cpal');
+    const isRaw = Number.isInteger(color);
+    let validFormat = true;
+    if (
+        (
+            isRaw && returnRaw
+        ) ||
+        color === 'currentColor'
+    ) {
+        return color;
+    } else if (typeof color === 'object') {
+        if (targetFormat == 'bgra') {
+            return color;
+        }
+        if (returnRaw) {
+            return bgraToRaw(color);
+        }
+    } else if(!isRaw && /^#([a-f0-9]{3}|[a-f0-9]{4}|[a-f0-9]{6}|[a-f0-9]{8})$/.test(color.trim())) {
```

Comment:
Uppercase are forbidden ? `[a-fA-F0-9]`

Reply:
Good catch, forgot the i modifier. I'll add a test for this as well.

Reply:
fixed